src/jit/compiler.h

```diff
@@ -2995,6 +2995,28 @@ class Compiler
                             CorInfoIntrinsics*    pIntrinsicID,
                             bool*                 isSpecialIntrinsic = nullptr);
     NamedIntrinsic lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method);
+
+#ifdef _TARGET_XARCH_
+    InstructionSet lookupHWIntrinsicISA(const char* className);
+    NamedIntrinsic lookupHWIntrinsic(const char* methodName, InstructionSet isa);
+    InstructionSet isaOfHWIntrinsic(NamedIntrinsic intrinsic);
+    GenTreePtr impX86HWIntrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HANDLE method);
```

Comment:
`GenTree*` instead of `GenTreePtr`

Reply:
Thanks, will do. Shall we always use `GenTree*` instead of `GenTreePtr`?

Reply:
I suggested this in other PRs and it seems that there is universal agreement about using `GenTree*` instead of `GenTreePtr`.

There is no `GenTreeCastPtr`, no `BasicBlockPtr`, no `CompilerPtr`. It's just `GenTreePtr`, an oddball.

Reply:
Got it.